tests/mapreduce/test.persisted.js

```diff
@@ -0,0 +1,687 @@
+            'theView' : storableViewObj
+          }
+        }, function (err) {
+          if (err) {
+            reject(err);
+          } else {
+            resolve('theViewDoc/theView');
+          }
+        });
+      });
+    }
+
+    before(function () {
+      // briefly increase the limit from 10 because we momentarily
+      // go over during these tests
+      PouchDB.setMaxListeners(30);
+    });
+
+    after(function () {
+      PouchDB.setMaxListeners(10);
+    });
+
+    beforeEach(function () {
+      Promise = PouchDB.utils.Promise;
+      return new PouchDB(dbName).destroy();
+    });
+    afterEach(function () {
+      return new PouchDB(dbName).destroy();
+    });
+
+    it('Test destroyed event on auxiliary db', function () {
+      var db = new PouchDB(dbName);
+      var rev;
+      return db.put({
+        _id: '_design/name',
+        views: {
+          name: {
+            map: function (doc) {
+              emit(doc.name);
+            }.toString()
+          }
+        }
+      }).then(function (res) {
+        rev = res.rev;
+        return db.bulkDocs([
+          {_id: 'foo', name: 'foo', title: 'yo'},
+          {_id: 'baz', name: 'bar', title: 'hey'},
+          {_id: 'bar', name: 'baz', title: 'wuzzup'}
+        ]);
+      }).then(function () {
+        return db.query('name');
+      }).then(function () {
+        return db.remove('_design/name', rev);
+      }).then(function () {
+        return db.viewCleanup();
+      }).then(function () {
+        return db.put({
+          _id: '_design/title',
+          views: {
+            title: {
+              map: function (doc) {
+                emit(doc.title);
+              }.toString()
+            }
+          }
+        });
+      }).then(function (res) {
+        rev = res.rev;
+      }).then(function () {
+        return db.query('title');
+      }).then(function () {
+        return db.remove('_design/title', rev);
+      }).then(function () {
+        return db.viewCleanup();
+      }).then(function () {
+        var views = ['name', 'title'];
+        return PouchDB.utils.Promise.all(views.map(function (view) {
+          return db.query(view).then(function (res) {
+            throw new Error('expected an error');
+          }, function (err) {
+            should.exist(err);
+          });
+        }));
+      }).then(function () {
+        return db.put({
+          _id: '_design/name',
+          views: {
+            name: {
+              map: function (doc) {
+                emit(doc.name);
+              }.toString()
+            }
+          }
+        }).then(function (res) {
+          rev = res.rev;
+          return db.query('name');
+        }).then(function (res) {
+          res.rows.map(function (row) {
+            return [row.id, row.key];
+          }).should.deep.equal([
+              ['baz', 'bar'],
+              ['bar', 'baz'],
+              ['foo', 'foo']
+            ]);
+        });
+      });
+    });
```

Comment:
This is a new test that I used to check the destroyed event on the dependent db. It succeeds before and after the fix, but I think it's always good to have more tests.